apps/hpc-ftsadmin/src/app/pages/add-edit/flow-edit.tsx

```diff
@@ -0,0 +1,1857 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { useParams, useLocation } from 'react-router-dom';
+import { Tooltip, TooltipProps, tooltipClasses } from '@mui/material';
+import { styled } from '@mui/material/styles';
+import dayjs from 'dayjs';
+import advancedFormat from 'dayjs/plugin/advancedFormat';
+import { isRight } from 'fp-ts/lib/Either';
+import * as ts from 'io-ts';
+import _, { values } from 'lodash';
+import {
+  C,
+  CLASSES,
+  combineClasses,
+  useDataLoader,
+  THEME,
+} from '@unocha/hpc-ui';
+import { categories, flows, forms, usageYears } from '@unocha/hpc-data';
+import { t } from '../../../i18n';
+import PageMeta from '../../components/page-meta';
+import { AppContext, getEnv } from '../../context';
+import tw from 'twin.macro';
+import FlowForm, {
+  FormValues,
+  ReportDetailType,
+  VersionDataType,
+  InputEntriesType,
+  AutoCompleteSelectionType,
+  ReportFileType,
+} from '../../components/flow-form';
+import {
+  FLOWS_FILTER_INITIAL_VALUES,
+  FlowsFilterValues,
+} from '../../components/filters/filter-flows-table';
+import { encodeFilters } from '../../utils/parse-filters';
+import {
+  getSearchKeyValues,
+  getFormValueFromCategory,
+  getFormValueFromFunding,
+  getNameOfFundingValue,
+  FundingSrcType,
+  FundingObjectType,
+  compareSelectValues,
+  IndividualFormValueType,
+  isArrayFieldOfInputEntries,
+  enumInputEntryVsSrcObjType,
+} from '../../utils/parse-form-values';
+
+dayjs.extend(advancedFormat);
+
+interface Props {
+  className?: string;
+  isEdit: boolean;
+}
+
+interface Participant {
+  name?: string;
+  email?: string;
+}
+
+const StyledLoader = tw(C.Loader)`
+mx-auto
+`;
+
+const StyledAnchor = tw.a`
+underline
+`;
+
+const StyledAnchorDiv = tw.div`
+text-2xl
+float-right
+`;
+
+const StyledSubTitle = tw.h3`
+text-2xl
+`;
+
+const StyledInactiveTitle = tw.span`
+text-gray-500
+`;
+
+const StyledTitle = tw.div`
+flex
+justify-between
+`;
+const StyledRestrictedCheckBox = tw.div`
+inline-flex
+items-center
+justify-center
+`;
+
+const YellowTooltip = styled(({ className, ...props }: TooltipProps) => (
+  <Tooltip {...props} classes={{ popper: className }} />
+))(({ theme }) => ({
+  [`& .${tooltipClasses.tooltip}`]: {
+    backgroundColor: '#ffec1a',
+    color: 'rgba(0, 0, 0, 0.87)',
+    boxShadow: theme.shadows[1],
+    fontSize: 11,
+  },
+}));
+
+const initialFormData = {
```

Comment:
Type initialFormData explicitly. TypeScript can't infer the types of empty arrays and many more.
```suggestion
const initialFormData: FormValues = {
```